File: hackerrank/easy/0024-circular-array-rotation/python/solution.py
## solution.py

from typing import List

def circular_array_rotation(a: List[int], k: int, queries: List[int]) -> List[int]:
    n = len(a)
    if n == 0:
        return []
    r = k % n
    return [a[(q - r) % n] for q in queries]


# --- demo run (from sample) ---
if __name__ == "__main__":
    a0 = [1, 2, 3]
    k0 = 2
    queries0 = [0, 1, 2]
    print(circular_array_rotation(a0, k0, queries0))  # [2, 3, 1]

File: hackerrank/easy/0024-circular-array-rotation/js/solution.mjs
// solution.mjs

export function circularArrayRotation(a, k, queries) {
  const n = a.length;
  if (n === 0) return [];
  const r = k % n;
  return queries.map(q => a[(q - r + n) % n]);
}

// --- demo run (from sample) ---
const a0 = [1, 2, 3];
const k0 = 2;
const queries0 = [0, 1, 2];
console.log(circularArrayRotation(a0, k0, queries0)); // [2, 3, 1]